tests/KubernetesClient.Tests/KubernetesClientConfigurationTests.cs

```diff
@@ -743,5 +743,29 @@ public void LoadInClusterNamespace()
                 Assert.Equal("some namespace", config.Namespace);
             }
         }
+
+        [Fact]
+        public void ChangeCurrentContext()
+        {
+            var k8sConfig =
+                KubernetesClientConfiguration.LoadKubeConfig(
+                    new FileInfo("assets/kubeconfig.yml"),
+                    false);
+            var cfg = KubernetesClientConfiguration.BuildConfigFromConfigObject(k8sConfig);
+            k8sConfig.CurrentContext = "federal-context";
+            var originalConfg = File.ReadAllText("assets/kubeconfig.yml");
+
+            var fileSystem = new MockFileSystem();
+
+            using (new FileUtils.InjectedFileSystem(fileSystem))
+            {
+                KubernetesClientConfiguration.SaveKubeConfig(k8sConfig, "assets/kubeconfig-Save.yml");
+                var newConf = File.ReadAllText("assets/kubeconfig-Save.yml");
+
+                Assert.True(File.Exists("assets/kubeconfig-Save.yml"));
+                Assert.Matches("current-context: federal-context", newConf);
+                Assert.Matches(@"cluster: horse-cluster[\n\r]{1,2}\W+user: green-user[\n\r]{1,2}\W+namespace: chisel-ns[\n\r]{1,2}\W+name: federal-context", newConf);
```

Comment:
This is brittle, I would suggest re-loading the KubeConfig file into a different configuration object and comparing the individual fields.

Reply:
Agreed. However, my end goal is to produce an equivalent file with only the expected changes, not just an object. So, I ultimately want to validate the text. My current logic would be valid if only comparing object properties, but the file is very different.